### DISCUSSION TITLE
Type Aware Clustering

### DISCUSSION THREAD

**Discussion by razorworks-hub** [quartznet/quartznet#947] [General] [answered]
Is it possible to have set of clustered instances, where each instance has different available job types, but still all read and write from a single database?

So in the following case you have an application with isolated process/contexts
HostContext    has a scheduler instance and knows of types JobA and JobB used purely for writing jobs and triggers.
ChildContextA has a scheduler instance to write or execute jobs of type JobA. Has no knowledge of ChildContextB or it's types
ChildContextB has a scheduler instance to write or execute jobs of  type JobB. Has no knowledge of ChildContextA or it's types

The reason I'd like to cluster this way. If any of the child contexts were to be offline (or just delayed) the remaining contexts could continue without issue. And if the offline context became operational it can just pick up it's outstanding triggers without them being stuck in the ERROR state. But from my testing if JobA is stored and the scheduler instance for ChildContextB has attempted to fire the trigger. It will result in the trigger being put into STATE_ERROR. Due to the type not being found

My thought being either a way to register types with a scheduler instance so it will only look for jobs from it's available types. Or way to allocate jobs themselves to specific scheduler instance ids when creating the job/trigger. (Not sure about this 2nd option because of the AUTO for scheduler ids).

**Comment by lahma** [comment, marked as answer]
Sounds like that you need a dedicated scheduler instance per context. This meaning that you have different scheduler instance name for them. This way they share the database, but not the jobs. Jobs and trigger are per scheduler instance name and instance id allows you to have multiple clustered contextA instances doing shared work - as long as they have different instance ids, like AUTO. [See configuration reference](https://www.quartz-scheduler.net/documentation/quartz-3.x/configuration/reference.html#main-configuration).

**Reply by razorworks-hub** [reply]
Thank you very much
